bridge/setu/processor/checkpoint.go

```diff
@@ -435,33 +444,35 @@ func (cp *CheckpointProcessor) fetchDividendAccountRoot() (accountroothash hmTyp
 
 // fetchLatestCheckpointTime - get latest checkpoint time from rootchain
 func (cp *CheckpointProcessor) getLatestCheckpointTime() (int64, error) {
-	currentHeaderNumber, err := cp.rootChainInstance.CurrentHeaderBlock(nil)
+	configParams, _ := util.GetConfigManagerParams(cp.cliCtx)
+	rootChainInstance, err := cp.contractConnector.GetRootChainInstance(configParams.ChainParams.RootChainAddress.EthAddress())
 	if err != nil {
-		cp.Logger.Error("Error while fetching current header block number", "error", err)
 		return 0, err
 	}
 
 	// fetch last header number
-	lastHeaderNumber := currentHeaderNumber.Uint64() // - helper.GetConfig().ChildBlockInterval
-	if lastHeaderNumber == 0 {
-		// First checkpoint required
+	lastHeaderNumber, err := cp.contractConnector.CurrentHeaderBlock(rootChainInstance)
+	if err != nil {
+		cp.Logger.Error("Error while fetching current header block number", "error", err)
 		return 0, err
 	}
+
 	// get big int header number
 	headerNumber := big.NewInt(0)
 	headerNumber.SetUint64(lastHeaderNumber)
```

Comment:
This is not used now since you are using `lastHeaderNumber` directly.

Reply:
Hmm, messed in merge with develop

Reply:
Okay.